### PR TITLE
feat: update action input type

### PR DESCRIPTION
**Pull request by mtruongsa**

### CODE REVIEW

File: packages/context/src/objects/objects.ts [hunk 1]
```diff
@@ -62,6 +62,11 @@ export type Obj = {
 
 export type ObjWithRoot = Obj & { rootObjectId: string };
 
+export type EffectiveObject = Obj & {
```
Comment (actong) [inline packages/context/src/objects/objects.ts:65]: Why is this needed?  I don't think there's any api that returns this.

File: packages/context/src/objects/objects.ts [hunk 2]
```diff
@@ -323,9 +332,9 @@ export type ActionInput = {
         when?: string;
         eq?: string | number | boolean;
     };
-    property?: {
-        id: string;
-    };
+    property?: InputParameter;
+    viewLayout?: ViewLayoutEntityReference;
+    documentMetadata?: DocumentMetadata;
```
Comment (actong) [inline packages/context/src/objects/objects.ts:337]: Is this the legacy `inputProperties`?  Are we still maintaining this?

**Comment by actong** [comment]
What are these changes for?

File: packages/context/src/objects/objects.ts (hunk 1 above)
Comment (mtruongsa) [reply packages/context/src/objects/objects.ts:65]: you are right. this is only used locally in Configuration repo to buildEffectiveObjects.

File: packages/context/src/objects/objects.ts (hunk 2 above)
Comment (mtruongsa) [reply packages/context/src/objects/objects.ts:337]: Yes it is legacy `inputProperties`. 
I am using this type `ActionInput` instead of the one in Configuration repo. The one called FormIOComponent is in the Configuration repo is removed. 
There are too many duplicated types between this repo and configuration repo. So I am removing all those duplicated ones in Configuration repo to use the ones in this repo as a single source of truth.

Comment (mtruongsa) [reply packages/context/src/objects/objects.ts:332]: Also, we are using `ActionInput` type when we convert form to formio component in Widget repo.

**Comment by mtruongsa** [comment]
> What are these changes for?

somehow my local always changes to this when I do git commit.

File: packages/context/src/objects/objects.ts [hunk 3]
```diff
@@ -278,6 +275,13 @@ export type ActionInputType =
     | 'Time'
     | 'User';
 
+export type DocumentType = 'Portal' | 'Private' | 'Public';
+
+export type DocumentMetadata = {
```
Comment (actong) [inline packages/context/src/objects/objects.ts:280]: Document metadata is open-ended.  As far as the platform is concerned, this is just `Record<string, string>`.

**Comment by actong** [comment]
> > What are these changes for?
> 
> somehow my local always changes to this when I do git commit for this repo.

I was asking about the PR as a whole.

File: packages/context/src/objects/objects.ts [hunk 4]
```diff
@@ -278,6 +275,8 @@ export type ActionInputType =
     | 'Time'
     | 'User';
 
+export type DocumentType = 'Portal' | 'Private' | 'Public';
```
Comment (actong) [inline packages/context/src/objects/objects.ts:278]: This is not needed either.

File: packages/context/src/objects/objects.ts [hunk 5]
```diff
@@ -323,9 +322,9 @@ export type ActionInput = {
         when?: string;
         eq?: string | number | boolean;
     };
-    property?: {
-        id: string;
-    };
+    property?: InputParameter;
+    viewLayout?: ViewLayoutEntityReference;
```
Comment (actong) [inline packages/context/src/objects/objects.ts:326]: Why is this being added to the legacy `inputProperties`?  When is this ever set?

Comment (mtruongsa) [reply packages/context/src/objects/objects.ts:326]: We have a function in Configuration and Widget repos to convert our form to form io components. The function returns a list of ActionInput components. To access `viewLayout` in those components, I need to add to `ActionInput` type.

**Review by actong — APPROVED** [review]
